scripts/west_commands/blobs.py

```diff
@@ -121,16 +131,58 @@ def list(self, args):
     def ensure_folder(self, path):
         path.parent.mkdir(parents=True, exist_ok=True)
 
-    def fetch_blob(self, url, path):
+    def fetch_blob(self, args, blob):
+        url = blob['url']
+        path = Path(blob['path'])
+
+        cached_blob = None
+        sha256 = blob['sha256']
+
+        # collect all cache dirs from west config and env variable
+        cache_dirs = args.cache_dirs or config.get('blobs', 'cache-dirs', fallback='')
+        auto_cache_dir = args.auto_cache or config.get('blobs', 'auto-cache', fallback=None)
+
+        # convert all cache dirs to Path
+        cache_paths = [Path(p) for p in cache_dirs.split(';') if p and Path(p).exists()]
+        if auto_cache_dir:
+            cache_paths.append(Path(auto_cache_dir))
+
+        # check if the blob is already cached
+        filenames = [f'{path.name}.{sha256}', path.name]
+        for cache_path in cache_paths:
+            for filename in filenames:
+                cached_file = cache_path / filename
+                self.dbg(f'Searching for blob in cache: {cached_file}')
+                status = zephyr_module.get_blob_status(cached_file.resolve(), sha256)
+                if status == zephyr_module.BLOB_PRESENT:
+                    cached_blob = cached_file
+                    break
+            if cached_blob:
+                break
+
         scheme = urlparse(url).scheme
-        self.dbg(f'Fetching {path} with {scheme}')
         import fetchers
         fetcher = fetchers.get_fetcher_cls(scheme)
 
         self.dbg(f'Found fetcher: {fetcher}')
         inst = fetcher()
+
+        # Fill the auto-cache if in use
+        if not cached_blob and auto_cache_dir:
```

Comment:
Why keep printing "Found fetcher" in the debug logs when it won't be used at all?

I agree with SonarQubeCloud that this function is getting a bit long; you can probably split it in two: 1. `get_blob()` that searches the cache and invokes `fetch_blob()` when not found. Also, probably reserve the term "fetch" for the network and use a different one for the cache to make the distinction clearer.